### PR TITLE
Docs: add legends shared content

### PR DESCRIPTION
**Pull request by imatwawana**
This PR adds and updates shared files for the legend options for visualizations. Two files have been added to cover the two most common sets of options (there are other visualizations that don't use shared content because they are outliers).  Using two shared files ensures two things:
1. Content remains consistent across visualizations that share the same options
2. Users reading about a visualization don't have to figure out which options don't apply to a specific visualization because they'll only be presented with the ones that apply.

This PR:
- Adds and updates shared files for legend content (2)
- Renames `legend-mode.md` to `legend-options-2.md`
- Renames `legend-options-2.md` to `legend-options-1.md`
- Adds shared files to appropriate visualizations
- Updates standalone legends content in heatmap and pie chart docs

**TO DO**

- [x] Technical/content review
- [x] Link style review (once tech review is done and changes implemented)
- [x] Update links

### CODE REVIEW

File: docs/sources/shared/visualizations/legend-options-1.md [hunk 1]
```diff
@@ -1,22 +1,29 @@
 ---
-title: Legend mode and legend placement
+title: Legend options
```
Comment (nmarrs) [inline docs/sources/shared/visualizations/legend-options-1.md:2]: should we add some internal comments to both of these legend option files to what visualizations they apply to / why there are two different legend option files? 😬

**Review by nmarrs — APPROVED** [review]
LGTM - just a quick question on adding some internal comments on nature / reasoning behind having two different legend option files 😬

**Comment by baldm0mma** [comment]
> LGTM - just a quick question on adding some internal comments on nature / reasoning behind having two different legend option files 😬

@imatwawana  Is there a reason we can't combine these 2 legend files? We could state which components adhere to what standards in a single doc file maybe?

File: docs/sources/shared/visualizations/legend-options-1.md (hunk 1 above)
Comment (imatwawana) [reply docs/sources/shared/visualizations/legend-options-1.md:2]: Good call. Will do!

**Comment by imatwawana** [comment]
I've created two files because this is a better experience for the reader/user. They don't have to worry about what applies to which visualization; they just see the options that apply. This just reduces load for the user.

**Comment by baldm0mma** [comment]
> I've created two files because this is a better experience for the reader/user. They don't have to worry about what applies to which visualization; they just see the options that apply. This just reduces load for the user. I'm also going to move the intro text into the shared file; in this case the intro text for the state timeline and status history is pretty different than what is for the rest of the visualizations that use the other shared file.

Ah, I see now. Awesome!

**Review by baldm0mma — APPROVED** [review]
LGTM

**Pull request merged**